src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,237 @@
+
+    .. csv-table:: Image csv example
+    :header: id, image_path
+    1194917,data/img_1.jpg
+    601566,data/img_2.jpg
+    554492,data/img_3.jpg
+    224125,data/img_4.jpg
+    2001716519,data/img_5.jpg
+    1257924,data/img_6.jpg
+    456490,data/img_7.jpg
+
+    .. csv-table:: Gallery Image csv example
+    :header: id, image_paths
+    8,data/db/img_1.jpg
+    10,data/db/img_2.jpg
+    12,data/db/img_3.jpg
+    """
+
+    def __init__(self,
+                 data_folder: str,
+                 matches_csv_path: str,
+                 img_list_csv_path: str,
+                 transform: Union[BasicTransform, BaseCompose],
+                 augment: Optional[Union[BasicTransform, BaseCompose]] = None,
+                 use_gallery: Optional[bool] = False,
+                 gallery_folder: Optional[str] = None,
+                 gallery_list_csv_path: Optional[str] = None,
+                 input_dtype: str = 'float32',
+                 input_column: str = 'image_path',
+                 grayscale: bool = False):
+        """
+        Args:
+            data_folder: Directory with all the images.
+            matches_csv_path: path to csv file where queries with their relevance scores are specified
+            img_list_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+                Id from matches csv are linked to id from img_list csv
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            use_gallery: If True, will use gallery data(non-relevant items)
+            gallery_folder: path to a folder with all gallery images (traversed recursively)
+            gallery_list_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+            input_dtype: Data type of of the torch tensors related to the image.
+            input_column: Name of the column that contains paths to images.
+            grayscale: If True, image will be read as grayscale otherwise as RGB.
+
+        Raises:
+            ValueError: if use_gallery True, but gallery_folder or gallery_list_csv_path is None
+        """
+        super().__init__(data_folder, transform, augment, input_dtype, input_column, grayscale)
+
+        self.__matches = pd.read_csv(self._data_folder / matches_csv_path,
+                                     usecols=['query', 'relevant', 'scores'],
+                                     dtype={'query': int, 'relevant': str, 'scores': str})
+
+        self.__img_paths = pd.read_csv(self._data_folder / img_list_csv_path,
+                                       usecols=['id', self._input_column],
+                                       dtype={'id': int, self._input_column: str})
+
+        self.__parse_match_csv()
```

Comment:
It's a bad practice to generate some class parameters in another method, without return these parameters as function return values. If you return that values you can write in documentation that exactly return your function, without it code would be unclear to read.